src/Microsoft.CodeQuality.Analyzers/Core/QualityGuidelines/MicrosoftQualityGuidelinesAnalyzersResources.resx

```diff
@@ -252,4 +252,10 @@
   <data name="AvoidPropertySelfAssignmentMessage" xml:space="preserve">
     <value>The property {0} should not be assigned to itself.</value>
   </data>
+  <data name="ReferingToObjectAndReassigningItInTheSameStatementMessage" xml:space="preserve">
+    <value>Object '{0}' is being referred to and reassigned in the same statement. You are at risk of referring to unintended object.</value>
```

Comment:
This might need rephrasing - tagging our documentation expert @gewarren for guidance.

Additionally, please also add a long description for the diagnostic which explains the issue in more detail. This resource string can be passed to the optional `description` parameter of the DiagnosticDescriptor constructor invocation in the analyzer.

Reply:
I've added description, but I think it still needs some improvements. Could you give me some tips?

Reply:
I suggested:

Referencing and reassigning an object in the same statement is not recommended. After the value of the object is changed, its value is referenced. Since it's in the same statement, the referenced value is the previous value before the reassignment. Accordingly, the reassignment is lost and the object retains its previous value.

Let me know what you think.

Reply:
Not quite. Take a look at the flowing example:

![image](https://user-images.githubusercontent.com/6609929/64299851-5bd71b00-cf83-11e9-9f52-77af9a4dce77.png)

In `TestOne` first assignment from the left is lost, but reassignment of object `first` is done successfully.

Reply:
How about: `Assigning to a symbol and its member (field/property) in the same statement is not recommended. It is not clear if the member access was intended to use symbol's old value prior to the assignment or new value from the assignment in this statement. For clarity, consider splitting the assignments into separate statements.`